### PR TITLE
Universe Finance Connector

### PR DESCRIPTION
**Pull request by UniverseFinance**
- [x] Review and approve connector @pradyuman-verma
- [x] Deploy the contracts and comment the addresses.
- [x] Created resolver - [here](https://github.com/Instadapp/dsa-resolvers/pull/37)
- [x] Add to the dsa-connect - [here](https://github.com/Instadapp/dsa-connect/pull/102)
- [ ] Add to the docs - link the PR

### CODE REVIEW

File: contracts/mainnet/connectors/universeFinance/main.sol [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     * @param universeVault Universe Official Vault Address
+     * @param amountA Amount of tokenA
+     * @param amountB Amount of tokenB
+     * @param getIds ID to retrieve amountA and amountB
+     */
+    function deposit(
+        address universeVault,
+        uint256 amountA,
+        uint256 amountB,
+        uint256[] calldata getIds
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/universeFinance/main.sol]: Add setIds as well to set the amount of share tokens.

File: contracts/mainnet/connectors/universeFinance/main.sol [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     * @param universeVault Universe Official Vault Address
+     * @param share0 Amount of uToken0.
+     * @param share1 Amount of uToken1.
+     * @param setIds stores the amount of output tokens
+     */
+    function withdraw(
+        address universeVault,
+        uint256 share0,
+        uint256 share1,
+        uint256[] calldata setIds
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/universeFinance/main.sol:47]: Add getIds as well to get the share amounts

File: contracts/mainnet/connectors/universeFinance/helpers.sol [hunk 3]
```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.7.6;
+
+import {TokenInterface} from "../../common/interfaces.sol";
+import {DSMath} from "../../common/math.sol";
+import {Basic} from "../../common/basic.sol";
+
+import  "./interface.sol";
+
+abstract contract Helpers is DSMath, Basic {
```
Comment (pradyuman-verma) [inline contracts/mainnet/connectors/universeFinance/helpers.sol:9]:
```suggestion
 contract Helpers is DSMath, Basic {
```

File: contracts/mainnet/connectors/universeFinance/main.sol [hunk 4]
```diff
@@ -0,0 +1,57 @@
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import {TokenInterface} from "../../common/interfaces.sol";
+import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+
+abstract contract UniverseFinanceConnect is Helpers, Events {
```
Comment (pradyuman-verma) [inline contracts/mainnet/connectors/universeFinance/main.sol:8]:
```suggestion
 contract UniverseFinanceConnect is Helpers, Events {
```

**Comment by UniverseFinance** [comment]
Connector  address:  0xd7a7296623c56f884b0a753ebd6653911f3986ca

**Comment by UniverseFinance** [comment]
Resolver address: 0xa7963898453c00b61cff8ce7e5b28c4e8bf1348f

**Comment by KaymasJain** [comment]
Connector:- "UNIVERSE-A" - 0xd7a7296623c56f884b0a753ebd6653911f3986ca

**Review by KaymasJain — APPROVED** [review]

**Pull request merged**